wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -640,11 +640,10 @@ impl PhysicalDeviceCapabilities {
             // Require `VK_EXT_descriptor_indexing` if one of the associated features was requested
             if requested_features.intersects(indexing_features()) {
                 extensions.push(vk::ExtDescriptorIndexingFn::name());
-
-                // Require `VK_KHR_maintenance3` due to it being a dependency
-                if self.effective_api_version < vk::API_VERSION_1_1 {
-                    extensions.push(vk::KhrMaintenance3Fn::name());
-                }
+            }
+            // Optional `VK_KHR_maintenance3`
+            if self.effective_api_version < vk::API_VERSION_1_1 {
```

Comment:
For this to be optional we need to check for `self.supports_extension(vk::KhrMaintenance3Fn::name())`.
Maybe we can move it just under `VK_KHR_maintenance2`.